### ISSUE TITLE
File.replace returns Invalid Changes data

### ISSUE THREAD

**Issue by dlanderson** [saltstack/salt#8301]
file.replace is not returning a dict of changes, so the highstate returns "Invalid Changes data". Should be a relatively easy fix:

```bash
----------
    State: - file
    Name:      /etc/ssh/sshd_config
    Function:  replace
        Result:    True
        Comment:   Changes were made
        Changes:   Invalid Changes data: --- 
+++ 
@@ -73,7 +73,7 @@
 # Allow client to pass locale environment variables
 AcceptEnv LANG LC_*
 
-Subsystem sftp /usr/lib/openssh/sftp-server
+Subsystem sftp internal-sftp
Match Group sftponly
        ChrootDirectory /home/sftpusers/%u
        X11Forwarding no
        AllowTcpForwarding no
        ForceCommand internal-sftp

 
 # Set this to 'yes' to enable PAM authentication, account processing,
 # and session processing. If this is enabled, PAM authentication will


Summary
------------
Succeeded: 1
Failed:    0
------------
Total:     1
```

**Comment by rallytime** [comment]
It looks to me like #21718 fixed this issue and should be resolved in 2015.5.0 or newer releases. Is anyone here still running into this issue?

**Comment by rallytime** [comment]
Since we didn't hear back, I am going to close this issue. If anyone runs into this bug again, let us know and we will be happy to reopen.
